core/lib/scheduler.rb

```diff
@@ -23,6 +22,8 @@ def self.search(params, schedules, schedule=[])
         end
       end
     else
+      schedule.average_start()
```

Comment:
no need to call these here

Reply:
Well don't i want to call those there because the schedule is populated, then the average values are calculated, and the schedule is added to an array.